Toolchain.mk

```diff
@@ -0,0 +1,174 @@
+BIN = $(CURDIR)/bin
+$(BIN):
+	mkdir -p $(BIN)
+
+PATH := $(BIN):$(PATH)
+
+M = $(shell printf "\033[34;1m▶\033[0m")
+
+CHAINSAW_VERSION = v0.2.10
+CONTROLLER_GEN_VERSION = v0.16.3
+CRDOC_VERSION = v0.6.4
+ENVTEST_VERSION = v0.20.4
+GOLANGCI_LINT_VERSION = v2.1.6
+HELM_DOCS_VERSION = v1.11.0
+HELM_VERSION = v3.16.2
+KIND_VERSION = v0.27.0
+KO_VERSION = v0.16.0
+KUSTOMIZE_VERSION = v5.6.0
+OPERATOR_SDK_VERSION = v1.32.0
+OPM_VERSION = v1.23.2
+YQ_VERSION = v4.35.2
+
+CHAINSAW := $(BIN)/chainsaw-$(CHAINSAW_VERSION)
+$(CHAINSAW): $(BIN)
+ifeq (, $(shell which $(CHAINSAW)))
+	$(info $(M) installing chainsaw)
+	@{ \
+	set -e ;\
+	OSTYPE=$(shell uname | awk '{print tolower($$0)}') && ARCH=$(shell go env GOARCH) && \
+	curl -sSLo $(CHAINSAW).tar.gz https://github.com/kyverno/chainsaw/releases/download/$(CHAINSAW_VERSION)/chainsaw_$${OSTYPE}_$${ARCH}.tar.gz && \
+	tar -zxvf $(CHAINSAW).tar.gz chainsaw && \
+	chmod +x chainsaw && \
```

Comment:
We can install a lot more of the tools using `go install` if the plan is to cache them in CI.
- [chainsaw](https://kyverno.github.io/chainsaw/0.1.1/install/#install-using-go-install)
- [kustomize](https://kubectl.docs.kubernetes.io/installation/kustomize/source/)
- [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installing-from-source)
- [golangci-lint](https://golangci-lint.run/welcome/install/#install-from-sources)
- [setup-envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/tools/setup-envtest)
- [yq](https://github.com/mikefarah/yq?tab=readme-ov-file#go-install)

Reply:
I intentionally moved away from go install, because it's slow (some tools take dozens of seconds to install), cache grows huge (when playing with bingo, I ended up having over 2Gi in compressed form, which was not scalable, which is slow to compress and to upload + is not that far from 10Gi cache limit on GitHub), it also matters how setup-go is configured (build cache for which tools gets included), and in which job the tools are built (does it suddenly become a pre-requisite? And if so, cache miss will cause somewhere towards a 10 minute job just for building things)

Reply:
That is some very convincing points to not use go install 😆

Thanks for taking a look